_docs-2/administration/upgrading.md

```diff
@@ -38,6 +56,43 @@ will then need to transfer the contents of the current individual files to this
 **Warning**: Upgrading a previously encrypted instance is not supported as the implementation
 and properties have changed.
 
+## Upgrading from 2.0 to 2.1
+
+Upgrading to Accumulo 2.1 is done by stopping Accumulo 2.0 and starting Accumulo 2.1.
+Before stopping Accumulo 2.0, install Accumulo 2.1 and configure it by following the 
+[Quick Start]({% durl getting-started/quickstart %}).
```

Comment:
I'm not sure installing 2.1 always makes sense to do before stopping 2.0. In some cases, the version may be managed by an RPM, which will replace 2.0 with 2.1. Also, this is misleading, because it suggests doing the quickstart to set up a new 2.1 cluster... but that's not what you would want to do if you're upgrading.

Reply:
The quick start reference was copied from the 1.8/9 to 2.0 section to be consistent.  I'll work on rewording, but there is a chicken / egg issue 

If installed from an rpm - would it be better to stop accumulo and then install the rpm or does it not matter?  I think it does, because if you want to use accumulo scripts to stop the running version, those scripts need to match the process names of that running version - but to get the config upgrade utility, you need 2.1 code.

Reply:
You could just as easily stop the 2.0 instance, install 2.1, run the upgrade utility, then start 2.1 instead of installing 2.1 prior to stopping 2.0.

The upgrade utility shouldn't run on a live 2.0 instance, but on a stopped one. So, I think what's important is that the optional config upgrade utility needs to be run after stopping the 2.0 instance, but before starting the 2.1 instance. I don't think it has anything to do with the order of installation relative to stopping the instance.

Reply:
I am trying to allow for installations that have a sensitivity to the length of downtime time required for the upgrade process to complete.  The time required to shutdown, run the upgrade utilities and then restart is the minimum that is necessary and the upgrade utilities runtime should be small (maybe a minute?)  and overall the time is relatively quick.

It is common for people to request the ability to do a rolling upgrade just to avoid the down-time - but with these changes that is not possible, the configuration must be converted before tablets are assigned to tservers and supporting both schemes in a running cluster is not just not supported, it flat out will not work. And that's without FATE and Thrift changes. Running 2.0 and 2.1 processes together is just not possible.

The time to distribute the new files and install the configuration could be as long or longer as the complete restart part of the process.  Because the distribution and configuration can be done with the cluster running, it does not need to contribute to the down time.  The key part being that the upgrade utilities need to be run with 2.1 code on a stopped 2.0 instance.

It all depends on the comfort level and the requirements of the user.  I do not want to imply that the cluster must be shutdown before installing the 2.1 software. But I also did not want to imply that someone needs to have a parallel installation of the software with 2.0 and 2.1 binaries simultaneously either.

My metal model is along the lines of preferring a  java-alternatives model where multiple versions are installed and available and selected with environment variables.  Probably a bad analogy because once running, we do not support switching back, but it also seems to be the best way to support restoring from a ZooKeeper dump and continuing with 2.0 if things went unexpectedly off-the rails and they did not want to extend the downtime while things get sorted.

Reply:
If you're trying to minimize the downtime, you could reword to achieve that goal, but in a way that's less misleading about the order requirement. Something along the lines of: "To minimize downtime, if your installation method allows for it, prepare the 2.1 binaries before you shut down 2.0, so you can upgrade and start 2.1 right away."

Reply:
> The key part being that the upgrade utilities need to be run with 2.1 code on a stopped 2.0 instance.

I think it would be good to emphasize this, perhaps word-for-word.